### ISSUE TITLE
Change the logic of FEMModel to copy the standard logic of Pytorch model built with nn.Module.

### ISSUE THREAD

**Issue by Solalspc** [AlexandreDabySeesaram/NeuROM#26]
At the moment, FEMModel doesn't look and feel like a model class.
It has no training attribute.
It's forward method gives the energy instead of enabling to evaluate the model.

It should bear a Model.energy method that was the former loss.
I propose it should have a training boolean attribute.
If training is True : forward compute another object based on IntegrationDomain (SeparatedDomain) ready to feed to the Model.energy.
If training is False : uses PointWiseInterpolator to interpolate.

I also propose it holds the FieldLayout.
And a Decomposition attribute. When the QoI is a field approximated by a Decomposition instance, Decomposition holds the structure for the approximation and the meshes, contexts and field manager methods.

**Comment by Solalspc** [comment]
A `NeuROMModel(field_layout, decomposition: TensorDecomposition, integration_domain, energy)` class is created to extend `FEMModel` in the tensor decomposition framework.
It should replace `FEMModel` in the long run. _If we want to create a model object for just one field, there should be a `TensorDecomposition` object for a single field named something like `NoDecomposition`._

**Q:** about the `forward` function, should it return the loss function of just the values of the model at the Gauss Points to be fed to the loss function then?

**A:** The choice is made to stick to the Pytorch model philosophy and return the model's values at the Gauss Points and then evaluate the loss function with those values. 
The forward is always an evaluation of the model seen as a function and the energy is kept distinct.

**Comment by jdichamp** [comment]
I'm not sure we have to stick to the classical model philosophy. I also don't like much having a split between training and inference within the `FEMModel` or `NeuROMModel`. We are doing FEM and ROM at the end of the day and this semantic should be prioritized IMHO compared to a ML semantic. If you want to train the nodal values of a field, then setup a training that does so. If you want to evaluate those fields at given points, then call appropriate evaluator. But I don't see why we should mix things. IMHO those models should be purely orchestrators: they call everything what is required for the loss to be computed (field interpolations basically) and the results are stored in the `FieldLayout`. For that reason, they also don't return anything. I agree however that currently, returning the loss might be confusing. Also, if you really want it, we can add an `evaluate()` but that won't be doing anything more than interpolating a field or evaluating a PGD decomposition which is the responsibility of other classes at the end of the day.

So, proposal of usage to maybe make it less confusing:
```python
# Define FEM model
model = FEMModel(
    mesh=mesh,
    field_layout=field_layout,
    integration_domain=domain,
    assemblies=[assembly_u, assembly_f],
)

#Define optimizer
optimizer = torch.optim.LBFGS(
    model.parameters(), lr=1e-1, max_iter=50, line_search_fn="strong_wolfe"
)

#Define closure
def closure():
    optimizer.zero_grad()
    model()  # interpolate fields into the layout (in place)
    loss = physics_loss() #Whatever loss you want 
    loss.backward(retain_graph=True)
    return loss

#Train
for _ in range(n_epochs):
    optimizer.step(closure)
```
